File: БР1.2/Михайлов_Юрий/labs/lab1/src/models/booking.entity.ts
import { BaseEntity, Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class Booking extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ type: 'int', nullable: false })
    property_id: number;

    @Column({ type: 'int', nullable: false })
    tenant_id: number;

    @Column({ type: 'date', nullable: false })
    start_date: string;

    @Column({ type: 'date', nullable: false })
    end_date: string;

    @Column({ type: 'varchar', length: 50, nullable: false })
    status: string;

    @Column({ type: 'text', nullable: false })
    details: string;
}

File: БР1.1/Захматов Юрий/labs/lab2/booking-service/src/main/java/com/petproject/bookingservice/entities/BookingEntity.java
package com.petproject.bookingservice.entities;


import com.petproject.bookingservice.enums.BookingStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Table(name = "booking")
@Entity
@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class BookingEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "renter_id", nullable = false)
    private Long renterId;


    @Column(name = "property_id", nullable = false)
    private Long propertyId;


    @Column(name = "payment_id")
    private Long paymentId;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BookingStatus bookingStatus = BookingStatus.PENDING;

    @Column(nullable = false)
    private Double totalPrice;

    private Integer guestsCount;

    private String details;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }
}
